.github/workflows/build_and_deploy.yml

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2022 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+name: Build, test and deploy docker images
+
+# The builds are triggered with pushes on the 'main' branch or with a pull
+# request against the 'main' branch
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+env:
+  base_version: 2
+  base_image_name: pika-ci-base
+  hip_version: 2
+  hip_image_name: pika-ci-hip
+
+jobs:
+  # Build base docker image
+  build_and_deploy:
+    # Cannot use the env variables in the name
+    name: Build and deploy images
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_PASS }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      # Base image #
+      # The deploy step is only triggered on push to main branch
+      - name: Build base image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: Dockerfile.base
+          push: ${{ github.event == 'push' }}
+          tags: ${{ secrets.DOCKER_USER }}/${{ env.base_image_name }}:latest,${{ secrets.DOCKER_USER }}/${{ env.base_image_name }}:${{ env.base_version }}
+      # HIP image #
+      # The deploy step is only triggered on push to main branch
+      - name: Build HIP image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: Dockerfile.hip
+          load: ${{ github.event != 'push' }}
+          push: ${{ github.event == 'push' }}
+          tags: ${{ secrets.DOCKER_USER }}/${{ env.hip_image_name }}:latest,${{ secrets.DOCKER_USER }}/${{ env.hip_image_name }}:${{ env.hip_version }}
+          build-args: BASE_IMAGE=${{ secrets.DOCKER_USER }}/${{ env.base_image_name }}:${{ env.base_version }}
```

Comment:
```suggestion
      - name: Build base image
        uses: docker/build-push-action@v2
        with:
          context: .
          file: Dockerfile.base
          push: ${{ github.event == 'push' }}
          tags: ${{ secrets.DOCKER_USER }}/${{ env.base_image_name }}:latest,${{ secrets.DOCKER_USER }}/${{ env.base_image_name }}:${{ env.base_version }}
          load: true
      # HIP image #
      # The deploy step is only triggered on push to main branch
      - name: Build HIP image
        uses: docker/build-push-action@v2
        with:
          context: .
          file: Dockerfile.hip
          push: ${{ github.event == 'push' }}
          tags: ${{ secrets.DOCKER_USER }}/${{ env.hip_image_name }}:latest,${{ secrets.DOCKER_USER }}/${{ env.hip_image_name }}:${{ env.hip_version }}
          build-args: BASE_IMAGE=${{ secrets.DOCKER_USER }}/${{ env.base_image_name }}:${{ env.base_version }}
```
I think?

Reply:
load and push can unfortunately not be set in the same step, will look into another solution